### PR TITLE
Update initial instruction for linux

### PR DESCRIPTION
**Pull request by khancyr**

### CODE REVIEW

**Comment by auturgy** [comment]
We have instuctions for forking and cloning elsewhere, so should link rather than repeat.
Also, telling people to clone the repo without forking first creates problems down track.  Much better to fork then clone.

**Comment by hamishwillee** [comment]
Re https://github.com/ArduPilot/ardupilot_wiki/pull/1248#issuecomment-409391259

@auturgy FWIW, I don't have a problem with repeating the instructions (rather than linking) because it disturbs flow to go to to another document and because first time you come here, you don't want people finding it anything other than very easy.

But it might be worth defining this in one place and [including where used](http://docutils.sourceforge.net/docs/ref/rst/directives.html#include). Not advocating, just suggesting.

I do agree that most people will have to fork, and it is worth covering both. Usually I show fork option, then have a note underneath "if you are just testing the current codeline but don't want to change anything you can clone the main repo" (or similar).

**Comment by khancyr** [comment]
Update from comments. And correct another mistake.

@auturgy we should probably advice to use fork but we don't do it in other place for now ...

I agree that the multiplication of the same instruction isn't that good. @hamishwillee suggestion is good ! 
Anyway, the building part should be redone, as it is badly done ...
I am just pushing some correctly from interns trying to install ArduPilot and getting stuck because instructions aren't clear enough for begginers ...

File: dev/source/docs/building-setup-linux.rst [hunk 1]
```diff
@@ -9,11 +9,26 @@ This article shows how to setup your build environment on Linux/Ubuntu machines.
...
+Download ArduPilot
+------------------
+
+If you don't have a copy of the ArduPilot git repository then open a
+terminal and run:
 
 ::
 
-    Tools/scripts/install-prereqs-ubuntu.sh -y
+    git clone git://github.com/ArduPilot/ardupilot.git
```
[the remark this answers is not in the document]
Comment (khancyr) [reply dev/source/docs/building-setup-linux.rst]: looks like the --recurse / --recursive-submodule isn't the same for all git version ... I prefer to keep the old way for now

File: dev/source/docs/building-setup-linux.rst [hunk 2]
```diff
@@ -9,11 +9,26 @@ This article shows how to setup your build environment on Linux/Ubuntu machines.
...
+Install some required packages
+------------------------------
+
+If you are on a debian based system (such as Ubuntu or Mint), we provide `a script <https://github.com/ArduPilot/ardupilot/blob/master/Tools/scripts/install-prereqs-ubuntu.sh>`__ that will do it for you :
+
+    - cd to the directory you :ref:`cloned <git-clone>` ArduPilot into.
+    - cd to the Tools/scripts directory ``cd Tools/scripts``
+    - run the script ``./install-prereqs-ubuntu.sh``.  You will be asked for your Ubuntu root password.  Respond with "Y" when it asks if you wish to install a package.
 
 Also follow the `MAVProxy's install instructions <https://ardupilot.github.io/MAVProxy/html/getting_started/download_and_installation.html#linux>`__ if you plan to use the simulator.
```
[the remark this answers is not in the document]
Comment (khancyr) [reply dev/source/docs/building-setup-linux.rst]: right

File: dev/source/docs/building-setup-linux.rst [hunk 3]
```diff
@@ -62,7 +78,7 @@ You need the specific gcc-arm cross-compiler linked above. You need to unpack it
 and then add the bin directory from the tarball to your $PATH by editing
 the $HOME/.bashrc file and adding a line like this to the end:
 
-``export PATH=$PATH:/home/your_username/bin/gcc-arm-none-eabi-4_9-2015q3/bin``
+``export PATH=$PATH:/home/your_username/where_you_unpack_the_toolchain/gcc-arm-none-eabi-4_9-2015q3/bin``
```
[the remark this answers is not in the document]
Comment (khancyr) [reply dev/source/docs/building-setup-linux.rst]: hum won't work as we ask to manually edit the bashrc

File: dev/source/docs/building-setup-linux.rst [hunk 4]
```diff
@@ -24,6 +39,7 @@ Reload the path (log-out and log-in to make permanent):
     . ~/.profile
 
 Now you should be able to build with waf as described in `BUILD.md <https://github.com/ArduPilot/ardupilot/blob/master/BUILD.md>`__.
+For more explanations and step-by-step instructions, please read :ref:`Setting up SITL on Linux <setting-up-sitl-on-linux>`.
```
[the remark this answers is not in the document]
Comment (khancyr) [reply dev/source/docs/building-setup-linux.rst]: ok changed

File: dev/source/docs/building-setup-linux.rst [hunk 5]
```diff
@@ -9,11 +9,26 @@ This article shows how to setup your build environment on Linux/Ubuntu machines.
...
+    git submodule update --init --recursive
+
+Install some required packages
+------------------------------
+
+If you are on a debian based system (such as Ubuntu or Mint), we provide `a script <https://github.com/ArduPilot/ardupilot/blob/master/Tools/scripts/install-prereqs-ubuntu.sh>`__ that will do it for you :
+
+    - cd to the directory you :ref:`cloned <git-clone>` ArduPilot into.
+    - cd to the Tools/scripts directory ``cd Tools/scripts``
+    - run the script ``./install-prereqs-ubuntu.sh``.  You will be asked for your Ubuntu root password.  Respond with "Y" when it asks if you wish to install a package.
```
[the remark this answers is not in the document]
Comment (khancyr) [reply dev/source/docs/building-setup-linux.rst]: done

**Comment by khancyr** [comment]
@peterbarker Thanks for comments ... I tried to update it. 
A big rewrite will be necessary, but out of the scope for now. I just want to patch this quick to prevent easy error.

**Comment by auturgy** [comment]
Thanks Pierre!

**Pull request merged**